dot/peerset/peerset.go

```diff
@@ -423,7 +424,16 @@ func (ps *PeerSet) allocSlots(setIdx int) error {
 
 		n := peerState.nodes[peerID]
 		if n.reputation < BannedThresholdValue {
-			logger.Critical("highest rated peer is below bannedThresholdValue")
+			/*
+				If our highest not connect peer is below threshold and we have no connections this is a problem.
+				However, if we have peers we are still connected to then this is not a big deal. For example,
+				if we have 10 peers in our set and are connected to 9 while 1 we are not connected because
+				they are below threshold, then our highestNotConnectedPeer is this one peer, which is ok
+				and we should just break
+			*/
+			if len(peerState.sortedPeers(setIdx)) == 0 {
```

Comment:
can we make it a single condition, like `low reputation AND no peers connected`?
```suggestion
			if n.reputation < BannedThresholdValue && len(peerState.sortedPeers(setIdx)) == 0 {
```

Reply:
Here we cannot, since regardless if the reputation is below the threshold we need to break. Youll notice after the connected peer condition we break. The other check is just for the log